juju/juju
# Add show-operation CLI and facade

### Checklist


There's a new API to add when actions work is integrated.

----

## Description of change

Add the show-operation CLI and backend.
The existing Operations() method was renamed to ListOperations().
There was also a driveby fix to timeout handling in show-task. The show-operation implementation is based on what was already done for show-task.

## QA steps

run an action on several units
run juju show-operation <id> before the operation has finished and see partial results
run juju show-operation --watch to see that it blocks until completion
run juju show-operation --wait 1s to see that it times out
run juju show-operation on a completed task to see full results
